package/MDAnalysis/coordinates/XDR.py

```diff
@@ -114,7 +130,8 @@ def _read_offsets(self, store=False):
             size = getsize(self.filename)
             try:
                 np.savez(offsets_filename(self.filename),
-                         offsets=offsets, size=size, ctime=ctime)
+                         offsets=offsets, size=size, ctime=ctime,
+                         n_atoms=self.n_atoms)
```

Comment:
We should be saving `self._xdr.n_atoms` instead of `self.n_atoms`, since the latter changes depending on whether `sub` is used.
By saving `self._xdr.n_atoms` the offsets can be reused even if the user sometimes uses the `sub` functionality and sometimes doesn't.